### PR TITLE
Fix performance for CUDA >= 9.2 (master)

### PR DESCRIPTION
**Pull request by lukasm91**
- sets `GT_CONSTEXPR` to `constexpr` for nvcc 
- introduces `const_ref` to fix performance for CUDA >= 9.2 (`T` for small data types, `T const&` for large data types)

### CODE REVIEW

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgridlaunch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
do perftest

File: unit_tests/common/test_pair.cpp [hunk 1]
```diff
@@ -26,8 +26,8 @@ TEST(pair, get_rval_ref) {
     size_t val0 = 1;
     size_t val1 = 2;
 
-    EXPECT_TRUE(
-        std::is_rvalue_reference<decltype(gridtools::get<0>(gridtools::pair<size_t, size_t>{val0, val1}))>::value);
+    static_assert(
```
Comment (anstaf) [inline unit_tests/common/test_pair.cpp]: It looks like it is not the part of the contract. I  would just delete this check.

Comment (lukasm91) [reply unit_tests/common/test_pair.cpp]: done

**Pull request merged**